tests/test_sql_insert_writer.py

```
@@ -0,0 +1,295 @@
+                                                    destination='tab5',
+                                                    type_cast=True)
+    assert 'DEFAULT::date,  -- ==> datecol1' in result
+    assert 'DEFAULT::integer,  -- ==> intcol1' in result
+
+
+@pytest.mark.skipif(PG_CTL_MISSING, reason='PostgreSQL not installed locally')
+def test_type_casting_insert_from(pg_url):
+    result = sql_insert_writer.generate_from_tables(
+        db_url=pg_url,
+        destination='tab5',
+        sources=['tab5_all_text', ],
+        type_cast=True)
+    assert 'datecol1::date,  -- ==> datecol1' in result
+    assert 'intcol1::integer,  -- ==> intcol1' in result
+    assert 'col2  -- ==> col2' in result
+
+
+@pytest.mark.skip
+def omit_autoincrementing_primary_keys(pg_url):
+    assert False
+
+
+def test_command_line_interface(sqlite_url):
+    """Test the CLI."""
+    runner = CliRunner()
+
+    # test invoking VALUE insert, one tuple
+    result = runner.invoke(cli.main, ['tab1', '--db', sqlite_url])
+    assert result.exit_code == 0
+    assert 'tab1' in result.output
+
+    # test invoking VALUE insert, two tuples
+    result = runner.invoke(cli.main,
+                           ['tab1', '--tuples', 2, '--db', sqlite_url])
+    assert result.exit_code == 0
+    assert 'tab1' in result.output
+
+    # test inserting from one table
+    result = runner.invoke(cli.main, ['tab1', 'tab2', '--db', sqlite_url])
+    assert result.exit_code == 0
+    assert 'tab2' in result.output
+
+    # test inserting from three tables
+    result = runner.invoke(cli.main, ['tab1', 'tab2', 'tab3', 'tab4', '--db',
+                                      sqlite_url])
+    assert result.exit_code == 0
+    assert 'tab2' in result.output
+
+    # # test that using --tuples with source tables raises
+    # with pytest.raises(BadOptionUsage):
+    #     runner.invoke(cli.main, ['tab1', 'tab2', '--tuples', 2, '--db', sqlite_url])
```

Should this commented-out code be removed? I'm curious why it's commented-out, though...

---

Oh, right, I was having trouble with that - investigating now.  Hmm.  Click's test runner swallows errors and repackages them in a weird way.  Ok, fixing that.